tests/features/community_content/content_sharing.feature

```diff
@@ -157,6 +157,86 @@ Feature: Sharing content between collections
       | discussion   |
       | news         |
 
+  @javascript
+  Scenario Outline: Share/Unshare contextual link should vary per user roles in a group.
+    Given collections:
+      | title      | state     |
+      | Westeros   | validated |
+      | Essos city | validated |
+    And "<content type>" content:
+      | title       | collection | state     |
+      | Iron throne | Westeros   | validated |
+    Given users:
+      | Username       | E-mail                     | Roles     |
+      | Jamie Lanister | jamie.lanister@example.com | moderator |
+      | John Snow      | john.snow@example.com      |           |
+      | Arya Stark     | arya.stark@example.com     |           |
+    And the following collection user memberships:
+      | collection | user       | roles       |
+      | Westeros   | John snow  | facilitator |
+      | Essos city | John snow  | member      |
+      | Essos city | Arya Stark | facilitator |
+    When I am logged in as "John Snow"
+    And I click "Keep up to date"
+    Then I should see the contextual link "Share" in the "Iron throne" tile
+    But I should not see the contextual link "Unshare" in the "Iron throne" tile
+
+    When I am logged in as "Arya Stark"
+    And I click "Keep up to date"
+    Then I should see the contextual link "Share" in the "Iron throne" tile
+    And I should not see the contextual link "Unshare" in the "Iron throne" tile
+
+    When I am logged in as "John Snow"
```

Comment:
We can save this relogging by doing the previous visibility checks first as Arya Stark and then as John Snow.